File: src/Java/Algs4LibraryPractice/Accumulator/VisualAccumulator.java
package Java.Algs4LibraryPractice.Accumulator;

import edu.princeton.cs.algs4.StdDraw;
import edu.princeton.cs.algs4.StdRandom;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.nio.Buffer;
import java.util.Iterator;

public class VisualAccumulator {
    private int count;
    private double sum;

    public VisualAccumulator(int totalNumbers, double max) {
        StdDraw.setXscale(0, totalNumbers);
        StdDraw.setYscale(0.0, max);
        StdDraw.setPenRadius(0.005);
    }

    public void addDateValue(double value) {
        sum += value;
        count++;
        StdDraw.setPenColor(StdDraw.DARK_GRAY);
        StdDraw.point(count, value);
        StdDraw.setPenColor(StdDraw.RED);
        StdDraw.point(count, mean());
    }

    public double mean() {
        return sum / count;
    }

    public String toString() {
        return "Mean (" + count + " values): " + String.format("%7.5f", mean());
    }

    // TestClient
    public static void main(String[] args) {
        int trial = Integer.parseInt(args[0]);
        VisualAccumulator visualAccumulator = new VisualAccumulator(trial, 1.0);

        for (int i = 0; i < trial; i++) {
            visualAccumulator.addDateValue(StdRandom.uniform(0.0, 1.0));
        }

        System.out.println(visualAccumulator);
    }
}

File: src/Kotlin/Algs4LibarayPractice/VisualAccumulator.kt
package Kotlin.Algs4LibarayPractice

import edu.princeton.cs.algs4.StdDraw
import edu.princeton.cs.algs4.StdRandom

class VisualAccumulator(private var max: Double, private var trial: Int) {
    private var sum = 0.0
    private var count = 0
    init {
        StdDraw.setXscale(0.0, trial.toDouble())
        StdDraw.setYscale(0.0, max)
        StdDraw.setPenRadius(0.005)
    }

    fun addDataValue(value: Double) {
        sum += value
        count++;

        StdDraw.setPenColor(StdDraw.DARK_GRAY);
        StdDraw.point(count.toDouble(), value);
        StdDraw.setPenColor(StdDraw.RED);
        StdDraw.point(count.toDouble(), mean());
    }

    fun mean(): Double {
        return sum / count;
    }

    override fun toString(): String {
        return "Mean $count values: " + String.format("%7.5f", mean());
    }
}

fun main(args: Array<String>) {
    var trials: Int = 2000
    var visualAccumulator = VisualAccumulator(1.0, trials)

    for (i: Int in 1..trials) {
        visualAccumulator.addDataValue(StdRandom.uniform());
    }

    println(visualAccumulator);
}